frontend/projects/upgrade/src/app/core/experiment-design-stepper/experiment-design-stepper.service.ts

```diff
@@ -294,45 +302,76 @@ export class ExperimentDesignStepperService {
 
   createNewFactorialConditionTableData(designData: ExperimentFactorialDesignData): FactorialConditionTableRowData[] {
     const tableData: FactorialConditionTableRowData[] = [];
+    const requiredFactorialTableData = this.factorDataToConditions(designData.factors);
 
-    // currently this table will only support 2 factors due to design constraints
-    // this will need revisited to support more factors in this table
-
-    const factorOne = designData.factors[0];
-    const factorTwo = designData.factors[1];
-
-    factorOne.levels.forEach((factorOneLevel) => {
-      factorTwo.levels.forEach((factorTwoLevel) => {
-        const tableRow: FactorialConditionTableRowData = {
-          id: uuidv4(), // TODO: maybe not the right place?
-          levels: [
-            {
-              id: factorOneLevel.id,
-              name: factorOneLevel.name,
-            },
-            {
-              id: factorTwoLevel.id,
-              name: factorTwoLevel.name,
-            },
-          ],
-          condition: this.createFactorialAliasString(
-            factorOne.name,
-            factorOneLevel.name,
-            factorTwo.name,
-            factorTwoLevel.name
-          ),
-          alias: '',
-          weight: '0.0',
-          include: true,
-        };
+    requiredFactorialTableData.map((conditionData) => {
+      const conditionLevelsData = this.filterLevelsData(conditionData);
+      const conditions = this.createConditionString(conditionData);
 
-        tableData.push(tableRow);
-      });
+      const tableRow: FactorialConditionTableRowData = {
+        id: uuidv4(), // TODO: maybe not the right place?
+        levels: conditionLevelsData,
+        condition: conditions,
+        alias: '',
+        weight: '0.0',
+        include: true,
+      };
+      console.log("tableRow.condition.split(';').length");
```

Comment:
oops console logs...

Reply:
Oops sorry missed that, it's removed now